tutorials/truffle-dev-life-cycle.md

```diff
@@ -0,0 +1,538 @@
+
+After our first test, we can go ahead and the following test:
+
+```js
+  it("should list a new Dizzy Dragon NFT", async () => {
+    // Access the logs of the mint transaction
+    const nftMintedLog = mintedNft.logs[2];
+    // Set the price for the NFT to 1 ether
+    const price = await web3.utils.toWei("1", "ether");
+    const nftAddress = nftMintedLog.address;
+    const tokenId = nftMintedLog.args[0]
+
+    // Call the listNft function of the NftMarketplace contract with the
+    // address of the NFT, the token ID, and the price
+    const listResult = await nftMarketplace.listNft(
+      nftAddress,
+      tokenId,
+      price
+    );
+    
+    const event = listResult.logs[0].event;
+    const seller = listResult.logs[0].args[0];
+    const tokenId = listResult.logs[0].args[2];
+
+    // Use Mocha's assert to test that the NftListed event was emitted
+    // with the correct arguments for the seller and token ID
+    assert.equal(event, 'NftListed')
+    assert.equal(seller, accounts[0])
+    assert.equal(tokenId.toNumber(), nftMintedLog.args[0].toNumber())
+  });
+```
+
+Again, you can run the tests to make sure that the tests pass as expected.
+
+### Test Purchasing NFTs {: #test-purchasing-nfts }
+
+Finally, let's test that an NFT on our marketplace can be purchased using the `purchaseNft` function of the `NftMarketplace` contract. Similarly to our previous tests, we'll use the event logs to test that the `NftPurchased` event has been emitted along with the correct state variables such as the buyer and token ID.
+
+Let's add the following test:
+
+```js
+  it("should buy a new Dizzy Dragon NFT", async () => {
+    const nftMintedLog = mintedNft.logs[2];
+
+    // List the NFT first
+    const price = await web3.utils.toWei("1", "ether");
+    const nftAddress = nftMintedLog.address;
+    const tokenId = nftMintedLog.args[0]
+    await nftMarketplace.listNft(
+      nftAddress,
+      tokenId,
+      price
+    );
+    
+    // Purchase the NFT using the purchaseNft function and passing in the
+    // address of the NFT and the token ID. We'll also send a payment along
+    // for the asking price of the NFT
+    const purchaseNft = await nftMarketplace.purchaseNft(
+        nftAddress,
+        tokenId,
+        {value: price}
+    );
+
+    const event = purchaseNft.logs[0].event;
+    const buyer = purchaseNft.logs[0].args[0];
+    const tokenId = listResult.logs[0].args[2];
+
+    // Use Mocha's assert to test that the NftPurchased event was emitted
+    // with the correct argument for the buyer and token ID
+    assert.equal(event, 'NftPurchased')
+    assert.equal(buyer, accounts[0])
+    assert.equal(tokenId.toNumber(), nftMintedLog.args[0].toNumber())
+  });
+```
+
+That's it for the tests! To run them all, go ahead and run:
+
+```
+npx truffle test --network dev
+```
+
+!!! challenge
```

Comment:
Maybe we can have the challenge box be colored with a Moonbeam brand collor? Purple?